src/test/java/com/vinyl/DeleteProductFromCartTest.java

```diff
@@ -0,0 +1,133 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.model.Cart;
+import com.vinyl.model.Product;
+import com.vinyl.model.ProductCart;
+import com.vinyl.model.User;
+import com.vinyl.repository.CartRepository;
+import com.vinyl.repository.ProductCartRepository;
+import com.vinyl.repository.TokenRepository;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.util.Lists;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+public class DeleteProductFromCartTest extends LoggedInBaseIntegration {
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    private ProductCartRepository productCartRepository;
+
+    @Autowired
+    private TokenRepository tokenRepository;
+
+    private Product product;
+
+    private Cart cart;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
+        cart = cartSetup();
+    }
+
+    @Override
+    String getUrl() {
+        return "/users/" + user.getId() + "/" + product.getId();
+    }
+
+    @Test
+    public void testWhenUserIsLoggedInAndCartExists() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        Assertions.assertThat(productCartRepository.findByProductAndCart(product, cart)).isNull();
+    }
+
+    @Test
+    public void testWhenUserIsLoggedInAndCartDoesNotExist() {
+        deleteCart();
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenUserIsLoggedInAndTriesToDeleteFromAnotherCart() {
+        User otherUser = createUser("otheruser@email.com");
+        Cart otherCart = defaultEntitiesHelper.createCart(otherUser);
+        defaultEntitiesHelper.createProductCart(otherCart,product);
+        HttpHeaders headers = tokenHeaderHelper.setupToken(token.getHash());
+
+        ResponseEntity<?> response = trt.exchange("/users/" + otherUser.getId() + "/" + product.getId(), HttpMethod.POST, new HttpEntity<>(headers), Void.class);
```

Comment:
setup headers part and sending the request go together, move them together for all methods and see how you can extract a more general method, maybe an overloaded version of setUpHeaderAndGetTheResponse